### PR TITLE
nvme: Add fw-download warning message for FWUG alignment error

### PR DESCRIPTION
**Pull request by ikegami-t**
Check the firmware file size if not aligned with FWUG value.

### CODE REVIEW

**Comment by keithbusch** [comment]
Since you're already in this function, could you change the xfer initialization, "`.xfer       = 4096,`" to set it to 0 instead?

**Comment by ikegami-t** [comment]
Okay will do. Thank you.

**Comment by ikegami-t** [comment]
The default size was changed by the commit 8b3d29e before is the change to `0` from `4,096` okay for the issue resolved by the change in the past?

**Comment by ikegami-t** [comment]
If needed we can use `argconfig_parse_seen(opts, "xfer")` instead of the defautl value chaning I think.

**Comment by keithbusch** [comment]
> The default size was changed by the commit [8b3d29e](https://github.com/linux-nvme/nvme-cli/commit/8b3d29ed3ad8113e54f5675b764f39a251d38ebd) before is the change to `0` from `4,096` okay for the issue resolved by the change in the past?

Yep, it's all handled correctly now, there's no need to keep the default at 4k anymore. We'll override it appropriately if so. The "parse_seen()" method works too.

**Comment by ikegami-t** [comment]
Thanks for the confirmation and merging the changes. Understood the current situation for the handling.

File: nvme.c [hunk 1]
```diff
@@ -4872,7 +4872,7 @@ static int fw_download(int argc, char **argv, struct command *cmd, struct plugin
 	int err;
 	struct stat sb;
 	void *fw_buf;
-	struct nvme_id_ctrl ctrl;
+	struct nvme_id_ctrl ctrl = { 0 };
```
[the remark this answers is not in the document]
Comment (ikegami-t) [reply nvme.c:4875]: I see. Okay will do fix this.

[the remark this answers is not in the document]
Comment (keithbusch) [reply nvme.c:4875]: We shouldn't need to do that anymore. The dma alignment of nvme is dword (4-bytes). Kernel will bounce if it's not aligned.

[the remark this answers is not in the document]
Comment (igaw) [reply nvme.c:4875]: Good to know, I missed this update on the kenrel side.

[the remark this answers is not in the document]
Comment (ikegami-t) [reply nvme.c:4875]: Noted and just created a PR #2313 so as mentioned will close it.

**Pull request merged**